gwsumm/data/coherence.py

```diff
@@ -131,8 +131,38 @@ def _get_coherence_spectrogram(channel_pair, segments, config=None,
         sampling = None
 
     # initialize component lists if they don't exist yet
+    # store compnents in a backup variable
+    coherence_bkp = {}
     for ck in ckeys:
         globalv.COHERENCE_COMPONENTS.setdefault(ck, SpectrogramList())
+        coherence_bkp[ck] = globalv.COHERENCE_COMPONENTS[ck]
+
+    # When coherence components contain different segments,
+    # computing coherence for new segments can result in a
+    # ValueError traceback due to incompatible shapes.
+    # To prevent this issue, we collect segments from all components,
+    # clear them from the global variable, and then restore from the
+    # coherence_bkp only the data that the segments available in
+    # coherence all components.
+
+    # get the segment spans from all components
+    spans = [SegmentList([
+        spec.span for spec in globalv.COHERENCE_COMPONENTS[ck]
+    ]) for ck in ckeys]
+    # keep only the intersection of the segments
+    spans = reduce(operator.and_, spans).coalesce()
```

Comment:
Trying to take advantage of the `gwpy` object types, this is untested but I put it forth as a suggestion to help improve the code readability. You'll need to `import SegmentListDict` too:
```suggestion
    spans = SegmentListDict()
    for ck in keys:
        spans[ck] = SegmentList([spec.span for spec in globalv.COHERENCE_COMPONENTS[ck]])
    # keep only the intersection of the segments
    spans = spans.intersection(list(ckeys))
```

Reply:
It doesn't look equivalent. According to the [documentation](https://gwpy.github.io/docs/stable/api/gwpy.segments.SegmentListDict/#gwpy.segments.SegmentListDict.intersects): the intersect method "returns True if there exists a segmentlist in self that intersects the corresponding segmentlist in other; returns False otherwise.". In the code `spans` should be a list of segments.

Reply:
@iaraota Check out https://gwpy.github.io/docs/stable/api/gwpy.segments.SegmentListDict/#gwpy.segments.SegmentListDict.intersection

Reply:
Thank you, I was looking into the wrong method. Applied this suggestion in new commit.